src/api.py

```diff
@@ -0,0 +1,146 @@
+import os
+
+from fastapi import FastAPI
+from fastapi.responses import RedirectResponse, FileResponse
+from fastapi.staticfiles import StaticFiles
+
+from .charset import URLCharset
+from .codec import Codec
+from .database import ViteDB
+
+## CONSTANTS ##
+
+PROJECT_ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+STATIC_PATH  = os.path.join(PROJECT_ROOT, 'src', 'static')
+DB_PATH      = os.path.join(PROJECT_ROOT, 'data', 'vite.db')
+
+DOMAIN_NAME  = "http://vite.lol/"
+SHORT_URL    = DOMAIN_NAME[7:] # without the http://
+
+## CORE LOGIC ##
+
+url_charset = URLCharset(numeric=True, lowercase_ascii=True,
+                         uppercase_ascii=True, special=False)
+codec       = Codec(charset=url_charset)
+app         = FastAPI(docs_url="/docs/")
+
+# Create the database if it doesn't exist and the links table
+os.makedirs(os.path.dirname(DB_PATH), exist_ok=True)
+with ViteDB(DB_PATH) as db:
+    db.create_table(db.table_name, db.fields)
+    
+# "/static/" avoids collisions with a possible /static generated path in the future
+app.mount("/static/", StaticFiles(directory=STATIC_PATH), name="static")
+    
+## API ENDPOINTS ##
+
+@app.get("/")
+def read_root() -> dict:
+    return FileResponse(f"{STATIC_PATH}/index.html")
+
+@app.get("/encode")
+def encode_url(url: str) -> dict:
+    """Encodes an URL to a short string
+
+    Args:
+        url (str): The URL to encode
+
+    Returns:
+        dict: A JSON response containing the encoded URL
+    """
+        
+    if url == "":
+        return {"error": "No URL provided"}
+    
+    with ViteDB(DB_PATH) as db:
+        row_count = db.get_row_count()
+        
+    unique_id: int = row_count + 1
+    encoded_uid: str = codec.encode(unique_id)
+    
+    shortened_url: str = f"{DOMAIN_NAME}{encoded_uid}"
+    
+    with ViteDB(DB_PATH) as db:
+        db.insert_link(url)
+    
+    return {"url": shortened_url}
+
+
+@app.get("/decode")
+def decode_url(url: str) -> dict:
+    """Decodes a short string to its original URL
+
+    Args:
+        url (str): The URL to decode
+
+    Returns:
+        dict: A JSON response containing the original URL and the number of clicks
+    """
+        
+    if url.startswith(DOMAIN_NAME):
+        url = url[len(DOMAIN_NAME):]
+    elif url.startswith(SHORT_URL):
+        url = url[len(SHORT_URL):]
```

Comment:
use `ternary` for readability
```suggestion
clearUrl = url[len(DOMAIN_NAME):] if url.startswith(DOMAIN_NAME) else (url[len(SHORT_URL):] if url.startswith(SHORT_URL): else url);

```

Reply:
was changed to a regex 
```python
rf"{DOMAIN_NAME}|{SHORT_URL}"
``` 
in https://github.com/atyrode/vite/pull/12/commits/499bebb26a581ec3e1a80184db1c5c5d681f0310